src/social_integration.cpp

```diff
@@ -0,0 +1,239 @@
+
+		if (plugin->library.HasError()) {
+			plugin->external.state = SocialIntegrationPlugin::FAILED;
+
+			Debug(misc, 0, "[Social Integration: {}] Failed to load library: {}", basepath, plugin->library.GetLastError());
+			return false;
+		}
+
+		OpenTTD_SocialIntegration_v1_GetInfo getinfo_func = plugin->library.GetFunction("SocialIntegration_v1_GetInfo");
+		if (plugin->library.HasError()) {
+			plugin->external.state = SocialIntegrationPlugin::UNSUPPORTED_API;
+
+			Debug(misc, 0, "[Social Integration: {}] Failed to find symbol SocialPlugin_v1_GetInfo: {}", basepath, plugin->library.GetLastError());
+			return false;
+		}
+
+		OpenTTD_SocialIntegration_v1_Init init_func = plugin->library.GetFunction("SocialIntegration_v1_Init");
+		if (plugin->library.HasError()) {
+			plugin->external.state = SocialIntegrationPlugin::UNSUPPORTED_API;
+
+			Debug(misc, 0, "[Social Integration {}] Failed to find symbol SocialPlugin_v1_Init: {}", basepath, plugin->library.GetLastError());
+			return false;
+		}
+
+		getinfo_func(&plugin->plugin_info);
+		/* Setup the information for the outside world to see. */
+		plugin->external.social_platform = plugin->plugin_info.social_platform;
+		plugin->external.name = plugin->plugin_info.name;
+		plugin->external.version = plugin->plugin_info.version;
+
+		/* Lowercase the string for comparison. */
+		std::string lc_social_platform = plugin->plugin_info.social_platform;
+		strtolower(lc_social_platform);
+
+		/* Prevent more than one plugin for a certain Social Platform to be loaded, as that never ends well. */
+		if (_loaded_social_platform.find(lc_social_platform) != _loaded_social_platform.end()) {
+			plugin->external.state = SocialIntegrationPlugin::DUPLICATE;
+
+			Debug(misc, 0, "[Social Integration: {}] Another plugin for {} is already loaded", basepath, plugin->plugin_info.social_platform);
+			return false;
+		}
+		_loaded_social_platform.insert(lc_social_platform);
+
+		auto state = init_func(&plugin->plugin_api, &plugin->openttd_info);
+		switch (state) {
+			case OTTD_SOCIAL_INTEGRATION_V1_INIT_SUCCESS:
+				plugin->external.state = SocialIntegrationPlugin::RUNNING;
+
+				Debug(misc, 1, "[Social Plugin: {}] Loaded for {}: {} ({})", basepath, plugin->plugin_info.social_platform, plugin->plugin_info.name, plugin->plugin_info.version);
+				return true;
+
+			case OTTD_SOCIAL_INTEGRATION_V1_INIT_FAILED:
+				plugin->external.state = SocialIntegrationPlugin::FAILED;
+
+				Debug(misc, 0, "[Social Plugin: {}] Failed to initialize", basepath);
+				return false;
+
+			case OTTD_SOCIAL_INTEGRATION_V1_INIT_PLATFORM_NOT_RUNNING:
+				plugin->external.state = SocialIntegrationPlugin::PLATFORM_NOT_RUNNING;
+
+				Debug(misc, 0, "[Social Plugin: {}] Failed to initialize: {} is not running", basepath, plugin->plugin_info.social_platform);
+				return false;
+
+			default:
+				NOT_REACHED();
+		}
+	}
+};
+
+std::vector<SocialIntegrationPlugin *> &SocialIntegration::GetPlugins()
+{
+	return _external_plugins;
```

Comment:
Personally I would have just created a local vector here, let that be filled from `_plugins` and then let return value optimisation take care to not introduce an actual copy of the vector. 

The current implementation will also work just fine.

Reply:
You are not wrong; overoptimization on my side!